### PR TITLE
FOGL-2924: Segmentation fault while RPM package on rhel/centos

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-2924: Segmentation fault while installing foglamp-service-notification package on rhel/centos

### CODE REVIEW

**Comment by Singhal-Vaibhav** [comment]
Adding @ashish-jabble too in review since he coded the generalisation of make_deb and rpm.

File: make_rpm [hunk 1]
```diff
@@ -185,9 +185,16 @@ mkdir -p usr/local/foglamp
...
+	./requirements.sh
 fi
-rm -rf build; mkdir ./build; cd ./build; cmake ..; make)
+
+if [ -f build.sh ]; then
+	rm -rf build;
+	./build.sh
+else
+	rm -rf build; mkdir ./build; cd ./build; cmake ..; make
+fi)
```
Comment (ashish-jabble) [inline make_rpm]: As of now we have a single `service` available for FogLAMP. We should place direct steps here for rpm. Likely below
```
source scl_source enable devtoolset-7
rm -rf build; mkdir ./build; cd ./build; cmake ..; make
```

We may move above approach as a generalised way under `foglamp-pkg/service` in **future**. And make service package from there.

Comment (pintomax) [reply make_rpm]: @ashish-jabble The make_rpm calls build.sh which does the right job depending on CentOS/RHEL 7 or Debian.

That's why you don't see in make_rpm

**Comment by MarkRiddoch** [comment]
> The requirements.sh looks like an exact copy of FogLAMP's requirements.sh. Not sure why are we duplicating the code. Its a safe assumption that foglamp-service-notification will be installed and executed on a system where FogLAMP is already there (and hence FogLAMP's requirements.sh have already been satisfied)

It is possible to run this micro service on a different machine to the others, so therefore it is good to have the requirements.sh.

**Comment by MarkRiddoch** [comment]
> Adding @ashish-jabble too in review since he coded the generalisation of make_deb and rpm.

I don't believe the current make_rpm in the package repository can do other micro services, just the core and plugins. It would be good to move this to the packaging repository and have a scripts that could work for an arbitrary microservice.

**Comment by ashish-jabble** [comment]
> in principle fine with this, however let's raise a Jira to get the packaging moved to the packaging repository as some point in the near future

@MarkRiddoch FOGL-2943 added

**Pull request merged**